src/main/java/org/sqlite/jdbc3/JDBC3DatabaseMetaData.java

```diff
@@ -842,6 +842,8 @@ public ResultSet getColumnPrivileges(String c, String s, String t, String colPat
     protected static final Pattern TYPE_INTEGER = Pattern.compile(".*(INT|BOOL).*");
     protected static final Pattern TYPE_VARCHAR = Pattern.compile(".*(CHAR|CLOB|TEXT|BLOB).*");
     protected static final Pattern TYPE_FLOAT = Pattern.compile(".*(REAL|FLOA|DOUB|DEC|NUM).*");
+    protected static final Pattern TYPE_WITH_SIZE = Pattern.compile("^(.*)\\(\\s*\\d+\\s*(\\s*,\\s*\\d+\\s*)?\\)$");
```

Comment:
maybe using named capturing group would be clearer ? Adding a comment on what the regex does and why would be helpful too.